sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -525,14 +527,18 @@ private LinkSubscription<AmqpSendLink> getSubscription(String linkName, String e
 
         final Sender sender = session.sender(linkName);
         sender.setTarget(target);
-
-        final Source source = new Source();
-        sender.setSource(source);
         sender.setSenderSettleMode(SenderSettleMode.UNSETTLED);
 
+        final Source source = new Source();
         if (linkProperties != null && linkProperties.size() > 0) {
+            final String clientIdentifier = (String) linkProperties.get(CLIENT_IDENTIFIER);
+            if (!CoreUtils.isNullOrEmpty(clientIdentifier)) {
+                source.setAddress(clientIdentifier);
+                linkProperties.remove(CLIENT_IDENTIFIER);
```

Comment:
I was curious why we remove it here?

Reply:
> I was curious why we remove it here?

I referred the .net code. When we create a send link, we don't put identifier in link properties, but create a receive link we do. So I remove it here.

Reply:
I'd like to make it clear:
1. in receiving link, we add identifier to target address and link properties named `com.microsoft:receiver-name`
2. in sending link, we add identifier to source address
so do we need to add identifier to sending link's properties? if we need, what's the property name?

Reply:
ah, that's my mistake.   You're doing the right thing; I misread the code and comment.